File: EV/backend/app/models/content.py
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


SourceType = Literal["text", "txt", "pdf", "docx", "pptx", "url", "youtube", "image", "video", "audio"]


class RawContent(BaseModel):
    model_config = ConfigDict(extra="forbid")

    source_id: str
    source_type: SourceType
    title: str
    text: str = Field(min_length=1)
    metadata: dict[str, Any] = Field(default_factory=dict)


class Identity(BaseModel):
    title: str = ""
    content_type: str = ""
    source_description: str = ""


class Overview(BaseModel):
    summary: str = ""
    purpose: str = ""


class Entities(BaseModel):
    people: list[str] = Field(default_factory=list)
    organizations: list[str] = Field(default_factory=list)
    locations: list[str] = Field(default_factory=list)
    technologies: list[str] = Field(default_factory=list)


class Facts(BaseModel):
    claims: list[str] = Field(default_factory=list)
    statistics: list[str] = Field(default_factory=list)
    dates: list[str] = Field(default_factory=list)
    events: list[str] = Field(default_factory=list)


class Findings(BaseModel):
    key_findings: list[str] = Field(default_factory=list)
    risks: list[str] = Field(default_factory=list)
    opportunities: list[str] = Field(default_factory=list)
    implications: list[str] = Field(default_factory=list)


class Recommendations(BaseModel):
    recommendations: list[str] = Field(default_factory=list)


class Context(BaseModel):
    target_audience: str = ""
    tone: str = ""
    communication_objective: str = ""


class Evidence(BaseModel):
    source_reference: str = ""
    supporting_excerpt: str = ""


class ContentDNA(BaseModel):
    model_config = ConfigDict(extra="forbid")

    identity: Identity = Field(default_factory=Identity)
    overview: Overview = Field(default_factory=Overview)
    entities: Entities = Field(default_factory=Entities)
    facts: Facts = Field(default_factory=Facts)
    findings: Findings = Field(default_factory=Findings)
    recommendations: Recommendations = Field(default_factory=Recommendations)
    context: Context = Field(default_factory=Context)
    evidence: Evidence = Field(default_factory=Evidence)

File: EV/frontend/src/types/content.ts
export type SourceType = 'text' | 'txt' | 'pdf' | 'docx' | 'pptx' | 'url' | 'youtube' | 'image' | 'video' | 'audio'

export interface RawContent {
  source_id: string
  source_type: SourceType
  title: string
  text: string
  metadata: Record<string, unknown>
}

export interface ContentDNA {
  identity: { title: string; content_type: string; source_description: string }
  overview: { summary: string; purpose: string }
  entities: { people: string[]; organizations: string[]; locations: string[]; technologies: string[] }
  facts: { claims: string[]; statistics: string[]; dates: string[]; events: string[] }
  findings: { key_findings: string[]; risks: string[]; opportunities: string[]; implications: string[] }
  recommendations: { recommendations: string[] }
  context: { target_audience: string; tone: string; communication_objective: string }
  evidence: { source_reference: string; supporting_excerpt: string }
}

export interface SourceRecord {
  source: RawContent
  content_dna: ContentDNA
}

export interface SourceCreatedResponse {
  source_id: string
  source_type: SourceType
  content_dna: ContentDNA
}

export type ContentDNAPatch = Partial<{
  identity: Partial<ContentDNA['identity']>
  overview: Partial<ContentDNA['overview']>
  entities: Partial<ContentDNA['entities']>
  facts: Partial<ContentDNA['facts']>
  findings: Partial<ContentDNA['findings']>
  recommendations: Partial<ContentDNA['recommendations']>
  context: Partial<ContentDNA['context']>
  evidence: Partial<ContentDNA['evidence']>
}>
